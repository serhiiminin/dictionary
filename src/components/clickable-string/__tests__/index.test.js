import React from 'react';
import renderer from 'react-test-renderer';
import ClickableString from '..';

describe('Clickable string', () => {
  const event = { preventDefault: () => {} };
  const component = renderer.create(
    <ClickableString
      item='String'
      onClick={() => { event.preventDefault(); }}
    />
  );
  let tree = component.toJSON();

  test('render', () => {
    expect(tree).toMatchSnapshot();
  });

  test('delimiter', () => {
    tree.props.delimiter=',';
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
