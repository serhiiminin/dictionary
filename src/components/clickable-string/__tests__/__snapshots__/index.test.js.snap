// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clickable string delimiter 1`] = `
<a
  className="ClicableString-clickableWord-0-1-1"
  href="/"
  onClick={[Function]}
>
  String
</a>
`;

exports[`Clickable string render 1`] = `
<a
  className="ClicableString-clickableWord-0-1-1"
  href="/"
  onClick={[Function]}
>
  String
</a>
`;
